src/frontend/src/components/LogIn/index.tsx

```diff
@@ -1,61 +1,192 @@
 import React from 'react';
 import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
 import { AppState } from '../../redux/store';
 import { logIn } from '../../redux/system/actions';
-import Logo from '../Logo';
 import { SystemState } from '../../redux/system/types';
+import apiService, { Request } from '../../service/api-service';
+import validateEmail from '../../service/email-validation';
 import history from '../History';
-import { Link } from 'react-router-dom';
+import Logo from '../Logo';
 
 interface Props {
   system: SystemState;
   logIn: typeof logIn;
 }
-interface State {}
+
+interface State {
+  email: string;
+  password: string;
+  invalidCredentials: boolean;
+  invalidResponse: boolean;
+  invalidEmail: boolean;
+  missingPassword: boolean;
+}
 
 class LogIn extends React.Component<Props, State> {
+  public state: State = {
+    email: '',
+    password: '',
+    invalidCredentials: false,
+    invalidResponse: false,
+    invalidEmail: false,
+    missingPassword: false
+  };
+
   constructor(props: Props) {
     super(props);
     this.logInNow = this.logInNow.bind(this);
   }
 
-  public logInNow(event: React.SyntheticEvent) {
-    this.props.logIn();
+  public handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+  };
+
+  public handleSubmit = (event: React.BaseSyntheticEvent) => {
     event.preventDefault();
     event.stopPropagation();
-    history.push('/oeci');
+
+    // Standardize email-lowerCase and trim both form inputs
+    this.setState(
+      {
+        email: this.state.email.toLowerCase().trim(),
+        password: this.state.password.trim()
+      },
+      () => {
+        this.validateForm();
+      }
+    );
+  };
+
+  public validateForm() {
+    // validate email returns true for email input of: "_@_._" empty returns false
+    this.setState(
+      {
+        invalidEmail: !validateEmail(this.state.email),
+        missingPassword: this.state.password.length === 0
+      },
+      () => {
+        // if no errors are present, logInNow()
```

Comment:
If a method does login, which is kind of a big deal side effect, its name should reflect that-- so I think it'd be good to rename this method.

Reply:
Good point, I had thought since that specific functions task was to handle the form validation for the subsequent l`ogInNow` call that `validateForm` was the best descriptive but I have changed:

`validateForm` --> `logIn`

Reply:
I'm now realizing that the redux-action is called `logIn` coming from `this.props.logIn()` which could provide some confusion.

How do you feel about:
`validateLogIn()`

Reply:
@ForrestLonganecker yeah that's fine.